### PR TITLE
chore: abstract away number to TNumber

### PR DESCRIPTION
**Pull request by davidkaplanbitgo**
Implement the changes done with the command

```bash
git checkout 6.1.0
git diff 6890d97beeec85b43c337e55a2b5cfc7dd23b87d -- ts_src
```

into master (for the files that still remain).

[BG-41154](https://bitgoinc.atlassian.net/browse/BG-54755)

### CODE REVIEW

**Review by brandonblack — APPROVED** [review]

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]
now that our Buffer polyfill is fixed in the frontend, we should use the native `readUInt64*` methods again https://github.com/BitGo/bitcoinjs-lib/commit/3ddc07a59460c19e58af1120cce3aa62a2b41fe2

**Review by sangaman — APPROVED** [review]
LGTM otto's comments about Buffer polyfill notwithstanding, some of the generics logic was a bit hard to follow for me but I don't have any concrete suggestions on how to improve them.

Also I wanted to clarify that the main purpose of having `TNumber` in the first place here is to allow us to deal w/ doge-like currencies that deal with numbers larger than what `number` can represent, right?

**Comment by OttoAllmendinger** [comment]
> Also I wanted to clarify that the main purpose of having TNumber in the first place here is to allow us to deal w/ doge-like currencies that deal with numbers larger than what number can represent, right?

Correct. In a future release we should simply drop support for `number`, which currently only exists for backwards compatibility.

**Review by sangaman — APPROVED** [review]

**Review by brandonblack — APPROVED** [review]
